### ISSUE TITLE
yield with function doesn't work

### ISSUE THREAD

**Issue by grevius5** [godotengine/godot#34265]
**Godot version:**
3.2 beta 3
**OS/device including version:**
Linux Manjaro
**Issue description:**
In the documentation for the yield you can see that:
```
You can also use yield to wait for a function to finish:

func _ready():
    yield(do_something(), "completed")
    yield(do_something_else(), "completed")
    print("All functions are done!")

func do_something():
    print("Something is done!")

func do_something_else():
    print("Something else is done!")
```

but if you try to yield a function you will get that error:
```
First argument of yield() not of type object.
```
**Steps to reproduce:**
yield(print("ok"), "completed")

**Comment by dalexeev** [comment]
@grevius5 See the documentation.
```
GDScriptFunctionState yield( Object object=null, String signal="" )
```

`print("ok")` returns `null`. This is not an object.
To wait for a function to complete, you should use something like the following:

```gdscript
extends Node

func f():
    print('f 1')
    yield(get_tree().create_timer(3), 'timeout')
    print('f 2')

func _ready():
    print('_ready 1')
    yield(f(), 'completed')
    print('_ready 2')
```

See also #31494.

**Comment by grevius5** [comment]
@dalexeev so is wrong the example in the documentation because it calls simple functions with only a print

**Comment by dalexeev** [comment]
@grevius5 I tried to find the phrase on the Internet, but did not find an exact match. Could you provide a link?
![191211-1](https://user-images.githubusercontent.com/47700418/70629741-7fbd9300-1c3b-11ea-99ad-3bdf70d0395c.png)

[Official documentation](https://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_basics.html):
![191211-2](https://user-images.githubusercontent.com/47700418/70629746-82b88380-1c3b-11ea-8c71-40a4e5e1a2c8.png)

**Comment by grevius5** [comment]
I have this in my 3.2 beta 3 godot internal documentation
![godot 3-2 beta 3 doc](https://user-images.githubusercontent.com/16955149/70631378-6281c800-1c2d-11ea-87c6-c0473583a451.png)

**Comment by dalexeev** [comment]
Indeed, this example (apparently incorrect) was added recently. #33872
@zaksnet @akien-mga

**Comment by zaksnet** [comment]
Hello guys,
It seems that in order to yield the function you are yielding also needs to inherit from GDScriptFunctionState. So the documentation should be changed to :
```
_Ready():
    yield(do_something(), "completed")
    yield(do_something_else(), "completed")
    print("All functions are done!")

func do_something():
    yield(get_tree(), "idle_frame")
    print("Something is done!")

func do_something_else():
    yield(get_tree(), "idle_frame")
    print("Something else is done!")
```

**Comment by dalexeev** [comment]
> _Ready():
```
func _ready():
```
:smile:

**Comment by zaksnet** [comment]
> > _Ready():
> 
> ```
> func _ready():
> ```
> 
> 😄

Been losing my mind lately going c# to GDScript >.<

**Comment by dalexeev** [comment]
> func _Ready():

```
func _ready():
```
GDScript is a case-sensitive programming language!
Do not rush like that. :smile: :smile: :smile:

**Comment by Xrayez** [comment]
`await` should solve most of these incorrect usages in the future hopefully: #32034, #33224 (meaning if you do pass `null` to `await`, it should handle that without errors).

I've also written some docs on this in godotengine/godot-docs#2644.

**Comment by dalexeev** [comment]
Fixed in #34280.

**Comment by DmDerbin** [comment]
var list = []
func myFunc1():
	call_deferred("myFunc2")
	yield(self, "processing_completed")
	print("222")


func myFunc2():
	print("111")
	for key in list:
		print("list item")
	emit_signal("processing_completed")


>>> 111
>>> 222
